controllers/asosecret_controller_test.go

```diff
@@ -175,7 +252,7 @@ func TestASOSecretReconcile(t *testing.T) {
 				g.Expect(reconciler.Recorder.(*record.FakeRecorder).Events).To(Receive(ContainSubstring(tc.event)))
 			}
 			if tc.err != "" {
-				g.Expect(err).To(MatchError(tc.err))
+				g.Expect(err).To(MatchError(ContainSubstring(tc.err)))
```

Comment:
Is there a reason we're no longer trying to match the error message in this test?

Reply:
Yes, some error messages are composed using different layers that are not necessarily ASO related and might be changed independent of the ASO controller, thus affecting the ASO secret tests. By introducing `ContainSubstring` we are checking only for the string we know will be thrown by the ASO controller.